lib/internal/modules/esm/create_dynamic_module.js

```diff
@@ -27,6 +36,17 @@ import.meta.exports.${name} = {
 };`;
 }
 
+/**
+ * Creates a dynamic module with the given imports, exports, URL, and evaluate function.
+ * @param {string[]} imports - An array of imports.
+ * @param {string[]} exports - An array of exports.
+ * @param {string} [url=''] - The URL of the module.
+ * @param {(reflect: DynamicModuleReflect) => void} evaluate - The function to evaluate the module.
+ * @typedef {object} DynamicModuleReflect
+ * @property {string[]} imports - The imports of the module.
+ * @property {string[]} exports - The exports of the module.
+ * @property {(cb: (reflect: DynamicModuleReflect) => void) => void} onReady - Callback to evaluate the module.
```

Comment:
```suggestion
 * @property {(cb: (reflect: DynamicModuleReflect) => void) => void} onReady - Callback to evaluate the module.
 * @returns {{ module: ModuleWrap, reflect: DynamicModuleReflect}}
```

Reply:
The `ModuleWrap` isn’t resolvable (TypeScript treats it as `any`) so I’m not sure about this one; the inferred return type is the same as you have but with `module: any`. It _does_ infer that `module` is the instantiated `new ModuleWrap`, but because `ModuleWrap` itself is indecipherable by TypeScript (I think because it’s a native binding) then it just gets treated like any unresolved reference.